tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb

```diff
@@ -1,1267 +1,1228 @@
+    "colab_type": "code",
+    "id": "mpAv77JoD98c"
+   },
+   "outputs": [],
+   "source": [
+    "if not os.path.exists('scripts'):\n",
+    "  print(\"Downloading necessary scripts\")\n",
+    "  !mkdir -p scripts/speaker_tasks\n",
+    "  !wget -P scripts/speaker_tasks/ https://raw.githubusercontent.com/NVIDIA/NeMo/$BRANCH/scripts/speaker_tasks/filelist_to_manifest.py\n",
+    "!python {NEMO_ROOT}/scripts/speaker_tasks/filelist_to_manifest.py --filelist {data_dir}/an4/wav/an4_clstk/train_all.txt --id -2 --out {data_dir}/an4/wav/an4_clstk/all_manifest.json --split"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "5kPCmx5DHvY5"
+   },
+   "source": [
+    "Generate the list text file for the test folder and then convert it to a manifest."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "nMd24GVaFBwr"
+   },
+   "outputs": [],
+   "source": [
+    "!find {data_dir}/an4/wav/an4test_clstk  -iname \"*.wav\" > {data_dir}/an4/wav/an4test_clstk/test_all.txt\n",
+    "!python {NEMO_ROOT}/scripts/speaker_tasks/filelist_to_manifest.py --filelist {data_dir}/an4/wav/an4test_clstk/test_all.txt --id -2 --out {data_dir}/an4/wav/an4test_clstk/test.json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "H5FPmxUkGakD"
+   },
+   "source": [
+    "## Path to manifest files\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "vo-VnYPtJO_v"
+   },
+   "outputs": [],
+   "source": [
+    "train_manifest = os.path.join(data_dir,'an4/wav/an4_clstk/train.json')\n",
+    "validation_manifest = os.path.join(data_dir,'an4/wav/an4_clstk/dev.json')\n",
+    "test_manifest = os.path.join(data_dir,'an4/wav/an4_clstk/dev.json')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "KyDVdtjAL2__"
+   },
+   "source": [
+    "As the goal of most speaker-related systems is to get good speaker level embeddings that could help distinguish from\n",
+    "other speakers, we shall first train these embeddings in end-to-end\n",
+    "manner optimizing the [TitaNet](https://arxiv.org/pdf/2110.04410.pdf) model.\n",
+    "We modify the decoder to get these fixed-size embeddings irrespective of the length of the input audio."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "OJtU_GEdMUUo"
+   },
+   "source": [
+    "# Training\n",
+    "Import necessary packages"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note: All the following steps are just for explanation of each section, but one can use provided [training script](https://github.com/NVIDIA/NeMo/blob/main/examples/speaker_tasks/recognition/speaker_reco.py) to train the model from command line."
```

Comment:
use provided -> use the provided
to train the model from command line -> to launch training in the command line